api/app/signals/apps/api/v1/serializers/signal.py

```diff
@@ -332,46 +332,35 @@ class PublicSignalCreateSerializer(serializers.ModelSerializer):
     class Meta(object):
         model = Signal
         fields = (
-            'id',
-            'signal_id',
             'source',
             'text',
             'text_extra',
             'location',
             'category',
             'reporter',
-            'status',
-            'priority',
-            'type',
-            'created_at',
-            'updated_at',
             'incident_date_start',
             'incident_date_end',
-            'image',
-            'attachments',
             'extra_properties',
         )
-        read_only_fields = (
-            'id',
-            'signal_id',
-            'created_at',
-            'updated_at',
-            'status',
-            'image',
-            'attachments',
-        )
         extra_kwargs = {
             'id': {'label': 'ID'},
             'signal_id': {'label': 'SIGNAL_ID'},
             'source': {'validators': [SignalSourceValidator()]},
         }
 
-    def create(self, validated_data):
-        if validated_data.get('status') is not None:
-            raise serializers.ValidationError("Status cannot be set on initial creation")
-        if validated_data.get('type') is not None:
-            raise serializers.ValidationError("Type cannot be set on initial creation")
+    def validate(self, data):
+        """Make sure any extra data is rejected"""
+        if hasattr(self, 'initial_data'):
+            present_keys = set(self.initial_data)
+            allowed_keys = set(self.fields)
 
+            if present_keys - allowed_keys:
```

Comment:
Nice, never understood why DRF is not doing this by default

Reply:
Neither do I.

Reply:
neither do I